pkg/scripts/os_amzn.go

```diff
@@ -0,0 +1,258 @@
+{{- end }}
+
+{{- if .NODE_BINARIES_URL }}
+curl -L --output /tmp/k8s-binaries/node.tar.gz {{ .NODE_BINARIES_URL }}
+tar xvf node.tar.gz
+{{- end }}
+
+{{- if and .KUBELET .NODE_BINARIES_URL }}
+sudo install --owner=0 --group=0 --mode=0755 /tmp/k8s-binaries/kubernetes/node/bin/kubelet /opt/bin/kubelet
+sudo ln -sf /opt/bin/kubelet /usr/bin/
+rm /tmp/k8s-binaries/kubernetes/node/bin/kubelet
+
+cat <<EOF | sudo tee /etc/systemd/system/kubelet.service
+[Unit]
+Description=kubelet: The Kubernetes Node Agent
+Documentation=https://kubernetes.io/docs/home/
+Wants=network-online.target
+After=network-online.target
+
+[Service]
+ExecStart=/opt/bin/kubelet
+Restart=always
+StartLimitInterval=0
+RestartSec=10
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+sudo mkdir -p /etc/systemd/system/kubelet.service.d
+cat <<EOF | sudo tee /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
+[Service]
+Environment="KUBELET_KUBECONFIG_ARGS=--bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf --kubeconfig=/etc/kubernetes/kubelet.conf"
+Environment="KUBELET_CONFIG_ARGS=--config=/var/lib/kubelet/config.yaml"
+# This is a file that "kubeadm init" and "kubeadm join" generates at runtime, populating the KUBELET_KUBEADM_ARGS variable dynamically
+EnvironmentFile=-/var/lib/kubelet/kubeadm-flags.env
+# This is a file that the user can use for overrides of the kubelet args as a last resort. Preferably, the user should use
+# the .NodeRegistration.KubeletExtraArgs object in the configuration files instead. KUBELET_EXTRA_ARGS should be sourced from this file.
+EnvironmentFile=-/etc/default/kubelet
+ExecStart=
+ExecStart=/opt/bin/kubelet \$KUBELET_KUBECONFIG_ARGS \$KUBELET_CONFIG_ARGS \$KUBELET_KUBEADM_ARGS \$KUBELET_EXTRA_ARGS
+EOF
+{{- end }}
+
+{{- if and .KUBEADM .NODE_BINARIES_URL }}
+sudo install --owner=0 --group=0 --mode=0755 /tmp/k8s-binaries/kubernetes/node/bin/kubeadm /opt/bin/kubeadm
+sudo ln -sf /opt/bin/kubeadm /usr/bin/
+rm /tmp/k8s-binaries/kubernetes/node/bin/kubeadm
+{{- end }}
+
+{{- if and .KUBECTL .KUBECTL_URL }}
+curl -L --output /tmp/k8s-binaries/kubectl {{ .KUBECTL_URL }}
+sudo install --owner=0 --group=0 --mode=0755 /tmp/k8s-binaries/kubectl /opt/bin/kubectl
+sudo ln -sf /opt/bin/kubectl /usr/bin/
+rm /tmp/k8s-binaries/kubectl
+{{- end }}
+
+{{ if .USE_KUBERNETES_REPO }}
+{{- if or .FORCE .UPGRADE }}
+sudo yum versionlock delete kubelet kubeadm kubectl kubernetes-cni || true
+{{- end }}
+
+sudo yum install -y \
+{{- if .KUBELET }}
+	kubelet-{{ .KUBERNETES_VERSION }} \
+{{- end }}
+{{- if .KUBEADM }}
+	kubeadm-{{ .KUBERNETES_VERSION }} \
+{{- end }}
+{{- if .KUBECTL }}
+	kubectl-{{ .KUBERNETES_VERSION }} \
+{{- end }}
+	kubernetes-cni-{{ .KUBERNETES_CNI_VERSION }}
+sudo yum versionlock add kubelet kubeadm kubectl kubernetes-cni
+{{- end }}
+
+sudo systemctl daemon-reload
+sudo systemctl enable --now kubelet
+
+{{- if or .FORCE .KUBELET }}
+sudo systemctl restart kubelet
+{{- end }}
+`
+
+	removeBinariesAmazonLinuxScriptTemplate = `
```

Comment:
This script will not work well if we use packages. We should check if the Kubernetes repository is used (`USE_KUBERNETES_REPO`) and use `yum` to remove all relevant packages in that case.

Reply:
Yeah. Fixed.